actions/changed-files/check_changed_files.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+# THIS expects two environment variables to be present
+#  - INCLUDE_PATTERN
+#  - EXCLUDE_PATTERN
+
+set -ex
+
+if [[ -z ${GITHUB_BASE_REF} ]]; then
+  # this is not a pull request event, just get the previous commit
+  previous_sha=$(git rev-parse --verify 'HEAD^{commit}')
+else
+  previous_sha=${GITHUB_BASE_REF}
+fi
+CHANGED_FILES=$(git diff --name-only --diff-filter=ACMRT ${previous_sha}..HEAD )
```

Comment:
Might be worth adding `set -u` after `GITHUB_BASE_REF` is checked to error out if there's any unset variables.

Reply:
I have preferred the later approach.